pkg/controller/core/workload_controller.go

```diff
@@ -134,6 +135,20 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	for i, ps := range wl.Spec.PodSets {
+		if wl.Spec.PodSets[i].PodTemplateName != nil {
+			continue
+		}
+		var pt corev1.PodTemplate
+		if err := r.client.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("%s-%s", wl.Name, ps.Name), Namespace: wl.Namespace}, &pt); err != nil {
+			continue
+		}
+		wl.Spec.PodSets[i].PodTemplateName = &pt.Name
+		if err := r.client.Update(ctx, &wl); err != nil {
+			return ctrl.Result{}, client.IgnoreNotFound(err)
+		}
```

Comment:
I think we should  not push the template copy to the API, and only use the fully populated wl internally (pass it to cache , queue etc.)